Inquisition Rework Pt. 1: Discipline

Kregle opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->

## About The Pull Request
The start to an INITIAL Inquisitorial rework as apart of this document: https://docs.google.com/document/d/1Ya8Zc6CPnXPt9mL5i2DbTlIgNMfakUC-hy3-8bjjYCI/edit?tab=t.0

Inquisitor changes:
-  A skill buff for the Inquisitor in Firearms!
- Magical resistance and neck protection for Inquisitors wearing their silver Psycross!

Confessor changes:
- Reflavoured to be former heretics known as "Repentants", more dedicated to being muscle than spies!

- A new tutorial message:
"Repentants are former heretics who have been seized and indoctrinated by the Inquisition. These wretched souls now seek to redeem themselves through the sting of the whip and muscling for the Inquisition, leading other wayward souls down the same bloody path they were forced onto themselves. Due their indoctrination, they are often more zealously naive compared to their knowledgeable Masters."

- Added the Critical Resistance and Dodge Expert!

- Removed their Speed and Endurance bonuses- in exchange giving them bonuses to strength and endurance!

- Gives them a stun baton in place of their cudgel and whip!

- Gives them expert in whips!

- PQ requirement from 4 to 6

- Shunned races and up! (minus kobold and verminfolk)

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
To quote from the doc above:
The Inquisitor needs to be feared and respected for the role to actually work, both within the lore and mechanically by players. People should think twice about trying to get in the way of the Inquisitor. The Inquisitor is essentially an anti-antag, in that they oppose other antags, but can often be a severe detriment to the town itself. Therefore, the Inquisitor should be the most equipped to deal with indirect/direct antagonists, especially Liches, Vampires and Verevolfs. This would make the Keep/Church/Town consider getting rid of one of their best assets against the inhumen at the slightest inconvenience.

The Inquisitor needs some proper goons if he is to get anything done; the current setup has resulted in VERY few Confessor players. I would propose reflavouring/working the confessors to be more like a couple of Militant-Flagellants. These should be STRONG boys, the fellas that are cracking bone, clobbering heads and black bagging civvies. They should also be able to at least decently hold their own against the Inhumen threats. Considering the changes to the Inquisitor, the standards for Confessors should be raised as well.

For the specific points please see my document
[https://docs.google.com/document/d/1Ya8Zc6CPnXPt9mL5i2DbTlIgNMfakUC-hy3-8bjjYCI/edit?tab=t.0](url)
<!-- Argue for the merits of your changes and how they benefit the game, especially if they are controversial and/or far reaching. If you can't actually explain WHY what you are doing will improve the game, then it probably isn't good for the game in the first place. -->


## Proof of Testing (Required)
[https://youtu.be/HhH5sBc-oao](url)
[https://www.youtube.com/watch?v=oH0OYGhf9VQ](url)
<!-- Show proof of testing, screenshots or recordings for features, proof of compile for backend changes -->

Red-Honk commented:
I quite frankly think the stunbaton for each repentent to be more than a tad overkill, those are not round start given to literally anyone for a reason (if you don't count the Duke/Duchess master rod), it is simply put busted and expensive to purchase for a reason, I think that giving them a tad of extra coin to be able to combine the funds they were given for the Inquisition in way of being half way there to buy one more than reasonable than straight up giving a no fun allowed tool to them at the start.

Also, is repentent a made up term that is going to be used, or was the word repentant misspelled?

Kregle commented:
> I quite frankly think the stunbaton for each repentent to be more than a tad overkill, those are not round start given to literally anyone for a reason (if you don't count the Duke/Duchess master rod), it is simply put busted and expensive to purchase for a reason, I think that giving them a tad of extra coin to be able to combine the funds they were given for the Inquisition in way of being half way there to buy one more than reasonable than straight up giving a no fun allowed tool to them at the start.
> 
> Also, is repentent a made up term that is going to be used, or was the word repentant misspelled?

All watchmen receive a stun baton. I don't particularly see it as being total overkill considering the amount of counters it has. They're meant for taking civilians down non-lethally and can serve relatively well as lethal maces (better than the cudgel atleast.

There were indeed a few spelling mistakes lmao, I'll fix them once I get home from work.

Skyglazer commented:
Could we get the lore changes in first? I feel like the bumping elbows with the Inquisition that the Keep often has isn't because they aren't strong enough, but is because of the differences in faith and how secular the Inquisition is. If they were all Astratan, the keep/church/etc. wouldn't have ideological reasons to not really like them all that much, and the nuance is frankly just hard for players to handle.

Kregle commented:
> Could we get the lore changes in first? I feel like the bumping elbows with the Inquisition that the Keep often has isn't because they aren't strong enough, but is because of the differences in faith and how secular the Inquisition is. If they were all Astratan, the keep/church/etc. wouldn't have ideological reasons to not really like them all that much, and the nuance is frankly just hard for players to handle.

The Inquisition is still an antagonist role and is intended to have conflict with the Town. My plans aren't about removing the conflict entirely, it's about forcing the garrison to have to seriously consider turning a blind to the Inquisition's less savoury actions for the sake of the enemy of my enemy is my friend- as well as not pissing off a small but powerful faction. If conflict does happen to come about, it'll be slightly less of a one way murder-fest and the Keep will be losing their best asset against the Inhumen/Zizo. I don't think anything I've included in the doc is necessarily conflicting with changing the Inquisition to be Astratan if that's the lore team's plan (which I do not oppose). As someone who plays a fairly "secular" inquisitor (in that I don't go spouting that the Ten are false and what not) I find that I still have conflict with people due to what the role entails.

Kregle commented:
Resolved the spelling mistakes (curse coding at 12 am) and cleaned up a bit of the changes. Also changed the PQ of Repentants from 4 to 6 to reflect the document.

Gyrandola reviewed (commented):
You might consider keeping the confessor snitch theme and giving them the listening devices they have(?) on Azure instead of reworking them into boring brutes.  But this is a personal suggestion and I'll leave the confessor changes to lore maints.

Balance-wise, the firearms skill is probably whatever, but I'm not sure giving the inquisitor knight-level sword skills is appropriate, especially when they already have expert whips, crossbows, wrestling...

I like the anti-magic properties of the silver psydon necklace.

Overall though, This PR feels less like a rework and more like a pretty serious buff.

Kregle commented:
> You might consider keeping the confessor snitch theme and giving them the listening devices they have(?) on Azure instead of reworking them into boring brutes. But this is a personal suggestion and I'll leave the confessor changes to lore maints.

Personally I think the Inquisition needs dedicated thugs more than it needs spies. With my plans in mind, the spying role would instead be left to a informants amongst the population. I wouldn't consider the Repentants boring (of course I'm very biased) basing them on chrisitanity's flagellants with a militant twist. Being former heretics, I feel like that opens up a lot of roleplay for them. Considering this I might even open it up to shunned races, excluding verminfolk and kobolds.

> Balance-wise, the firearms skill is probably whatever, but I'm not sure giving the inquisitor knight-level sword skills is appropriate, especially when they already have expert whips, crossbows, wrestling...

I don't believe the firearms skill affects the runelock from my testing, but it would come in use if they somehow lost the runelock (their best piece of equipment aside from the silver psycross lmao) and had to use a regular gun instead. This may be an issue of skill bloat in other roles, but I personally feel that compared to other roles the Master level swords is fair. With this rework in mind, the Inquisitor I personally feel should be better with their main weapon than mercenaries, warrior adventurers, MAAs, etc. Considering the intense difference in arms and armor, if an Inquisitor whiffs their runeshot, they'd still be likely to lose to a knight or fully prepared bandit regardless. One of the main benefits - imo - regarding Inquisitor is how their stat spread affects their feint intent; being that they have a good bonus to intelligence, decent strength and perception and no malus to speed, enemies have a good chance of falling for their feint intents, opening them for a runelock or other shenanigans.

> I like the anti-magic properties of the silver psydon necklace.

Good, it was a pain to get to work lmao

> Overall though, This PR feels less like a rework and more like a pretty serious buff.

That's a fair assessment, though I should say this is only one part of my plans regarding this - with mayhaps the most controversial part being a whitelist for Inquisitor. Even everything set out in the document is only meant to be initial. I have some future plans to think out some ideas for subclasses for inquisitor and repentant/confessor, make a hoardmaster type system to reward inquisitors for extracting confessions, mayhaps even more torture methods. The reason I'm doing this so incrementally is that I'm not an experienced coder (this is my first PR for SS13) and the potential lore changes surrounding Inquisitor that have been mentioned above that may necessitate changes.

AukariBunn commented:
I believe the inquisitor should also have at least one actual student. Repentants are the swords, not the hand wielding that sword. The inquisitor should have a person yet to become an inqusitor in their retinue, who is also the second in command in the rockhillian inquisition.
Anyways they all should be given some proper armor, like besides the inquisitor themselves, the rest of the inqusition has the worst armor among all the town combat roles.

Speaking of some other suggestions:
Also I recently came up with an idea of spy ravens: mechanical magic-powered birds that provide a video feed to the operator, they can be used to spy on both the town folk and the nobility.
The inqusition should be able to get additional supplies from Kingsfield, something like the hoardmaster. It can basically be implemented through the HERMES.

RadiantFlash7 commented on code/modules/clothing/rogueclothes/neck/neck.dm:
```diff
@@ -258,9 +258,14 @@
 	name = "silver psycross"
 	desc = "Make no mistake, son of PSYDON, this amulet is as valuable as any blade in your crusade. Inhumen monsters and cultists shrink at the sight of silver, for it is the All-Father's blood made manifest. So hoist it high, scream his name, and bathe this world in blood so that it might be redeemed."
 	icon_state = "psicrossiron"
+	max_integrity = 300
```
I don't really think we should be giving the inquisition faith powered neck armor.
Pysdon's dead, They're fanatics not holy warriors.

Kregle replied:
That's fair enough, though I should say that I based the addition of armor on the dragonscale necklace. I should probably reflavour how it's stated in the activation text- but this meant to be less of a miracle wielding holy warrior type of thing and more of an Inquisitor using unique training to use silver's inherently divine nature (being the blood of Psydon) which still clearly has some power considering that silver is anathema to nite-creechers.

RadiantFlash7 commented on code/modules/clothing/rogueclothes/neck/neck.dm:
```diff
@@ -311,6 +316,28 @@
+	active_item = TRUE
+	if(HAS_TRAIT(user, TRAIT_ZEALOT))
+		to_chat(user, span_notice("I feel my faith in PSYDON protecting me!"))
+		ADD_TRAIT(user, TRAIT_ANTIMAGIC, TRAIT_GENERIC)
```
I also very specifically don't think we should give them constanti anti-magic. All of our antimagic sources are time limited, be it via a temporary antimagic effect followed by a cooldown; or taking a drug that provides brief antimagicy with harmful visual effects.

RadiantFlash7 commented on code/modules/jobs/job_types/roguetown/inquisition/confessor.dm:
```diff
@@ -3,8 +3,8 @@ This confessor is different to 'confessors', the people who are hit with the con
 /datum/job/roguetown/shepherd
-	title = "Confessor"
-	flag = CONFESSOR
+	title = "Repentant"
```
I have no real thoughts on the name changes for confessor. They're effectively the same, so I'd be fine with having it changed.

RadiantFlash7 commented on code/modules/jobs/job_types/roguetown/inquisition/confessor.dm:
```diff
@@ -61,13 +64,12 @@ Please do not confuse the two.
 		H.change_stat("intelligence", -1)
-		H.change_stat("endurance", 1)
-		H.change_stat("strength", 2)
-		H.change_stat("speed", 2)
```
I actually highly prefer this statblock change. Confessors being less speedy allows chances for heretics to flee (most towner roles don't get +2 speed), while keeping them very tanky and hard to get out of wrestling grip from.

RadiantFlash7 commented on code/modules/jobs/job_types/roguetown/inquisition/puritan.dm:
```diff
@@ -44,12 +46,12 @@
 	backpack_contents = list(/obj/item/storage/keyring/puritan = 1, /obj/item/rogueweapon/huntingknife/idagger/silver, /obj/item/lockpick = 1, /obj/item/clothing/head/roguetown/helmet/overseer/vice)
 	if(H.mind)
-		H.mind.adjust_skillrank(/datum/skill/combat/swords, 4, TRUE)
+		H.mind.adjust_skillrank(/datum/skill/combat/swords, 5, TRUE)
```
The inquisitor can stay at 4 swords honestly. We don't need them at master

RadiantFlash7 commented on code/modules/jobs/job_types/roguetown/inquisition/puritan.dm:
```diff
@@ -44,12 +46,12 @@
 		H.mind.adjust_skillrank(/datum/skill/combat/wrestling, 4, TRUE)
 		H.mind.adjust_skillrank(/datum/skill/combat/unarmed, 3, TRUE)
-		H.mind.adjust_skillrank(/datum/skill/combat/firearms, 3, TRUE)
+		H.mind.adjust_skillrank(/datum/skill/combat/firearms, 5, TRUE)
```
The inquisitors fire-arm is broken. Flat out, regardless of firearm skill it can fire instantly with full accuracy, dealing about 200 damage with a HIGH level of AP. Because of this, he doesn't really need his fire-arm skill increased- It can stay at 3.

RadiantFlash7 reviewed (changes requested):
All in all, A few things needing changes. T

- The inquisition shouldn't be getting armor powered by faith- Pysdon is quite litterally dead. It's why no psydon clerics exist. Either make an armored subtype of cross unique to them with some other explanation, or give them unique neck armor thats not a cross.
- Confessor changes are- fine. I actually like most of them.
- Inquisitor skill increase should be scrapped. Fairly un-needed.

When it comes to your future plans for the inquisition- I don't really like it key portions of it.

 We don't want to push the inquisition harder into fighting our major antagonists AND make them a whitelisted job. If anything, the inquisition should be oriented around the heretical townsfolk and cultists (Something a PR is working on reimplementing soon-ish).

Kregle commented:
> When it comes to your future plans for the inquisition- I don't really like it key portions of it.
> 
> We don't want to push the inquisition harder into fighting our major antagonists AND make them a whitelisted job. If anything, the inquisition should be oriented around the heretical townsfolk and cultists (Something a PR is working on reimplementing soon-ish).

I'll see if I can't get in contact with that bloke doing the heretic stuff. I'd absolutely love to work together with them to have synergy regarding that. 

I understand that there will be some opposition to the idea of adding a whitelist - I should say that I am opposed to whitelist other than Inquisitor, as it is a round impacting role that isn't required for a round to function - but I think that we'll see the quality of the roleplay coming out of Inquisitors improve, while also hopefully seeing the role taken a fair bit more seriously by those interacting with them, without the need to make them an absolute OP killing machine.

Kregle commented:
Added new sprites for the psycrosses and now all shunned races (not including kobold/verminfolk) to be Repentants.

Peter-Dolby commented:
I think the biggest thing the inquisitor needs before anything else is more pockets, and possibly changes to bullet-pouches and pistols to make them fit in containers more. They've just got so much shit they need to carry and no slots to carry it in. Then you can do the rest of it.

RadiantFlash7 commented:
Main point before I can approve the rest of this PR:
Inquisitor does not need a all the time magic immunity item.

Franky, I don't think they deserve one at all. They already hard counter 2 antags with silver, and have an instant win gun for every fight. But I'll allow it if it functions similar to a nullring. Either get that done at a minimum, or this PR won't be approved.

Kregle commented:
I'VE DONE IT!!! I made the silver psycross have an active time/cooldown, activated on right click exactly like the null magic ring. It's a fair bit better with the active time being double, and cooldown time being halved.

Also titles for Inquisitors and Inquisitrixs.
